File: 브론즈 1 (7문제)/2609_최대공약수와 최소공배수.c
// 설명 링크 : https://moon-gd.tistory.com/102
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>

int find_gcd(int num1, int num2);


int main() {
	int a, b;
	scanf("%d %d", &a, &b);

	int gcd = find_gcd(a, b), lcm = a * b / gcd;

	printf("%d\n", gcd);
	printf("%d", lcm);

	return 0;
}

int find_gcd(int num1, int num2) {
	int smaller_num = num1 < num2 ? num1 : num2;

	for (int i = smaller_num; i >= 1; i--) {
		if (num1 % i == 0 && num2 % i == 0) {
			return i;
		}
	}

	return 1;
}

File: 브론즈 1 (7문제)/2609_최대공약수와 최소공배수.py
# 설명 링크 : https://moon-gd.tistory.com/102
def find_gcd(a, b):
    smaller_one = min(a, b)

    for i in range(smaller_one, 0, -1):
        if a % i == 0 and b % i == 0:
            return i


num1, num2 = map(int, input().split())  # 입력 받기
gcd = find_gcd(num1, num2)  # 최대 공약수 구하기
lcm = num1 * num2 // gcd  # 최소 공배수 계산

# 결과 출력
print(gcd)
print(lcm)